tests/test_routing.py

```diff
@@ -521,6 +521,14 @@ def test_implicit_head():
                   '/post', method='HEAD')
 
 
+def test_pass_str_as_router_methods():
+    try:
```

Comment:
It took me two reads to get what you were trying to achieve. Please use https://pytest.org/latest/assert.html#assertions-about-expected-exceptions